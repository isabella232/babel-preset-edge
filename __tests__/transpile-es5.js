import buildPreset from "../src"

import { check, fixtures, titles } from "./core"

describe("Transpile: ES5", () => {
  const options = buildPreset(null, {
    sourceMaps: false,
    transpile: "es5"
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
