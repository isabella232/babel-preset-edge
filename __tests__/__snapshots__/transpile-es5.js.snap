// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transpile: ES5 alias 1`] = `
"import { Button } from \\"../../src/components/button/Button\\";
import { logx } from \\"../../src/testonly\\";
import \\"src/testonly\\";
logx(Button);"
`;

exports[`Transpile: ES5 alias.tsx 1`] = `
"import { Button } from \\"../../src/components/button/Button\\";
import { logx } from \\"../../src/testonly\\";
import 'src/testonly';
logx(Button);"
`;

exports[`Transpile: ES5 comments 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

function translate(value) {
  return _camelCase(value);
}"
`;

exports[`Transpile: ES5 deadcode 1`] = `
"var STATIC = false;
before();
after();"
`;

exports[`Transpile: ES5 es2015: classes 1`] = `
"import _inheritsLoose from \\"@babel/runtime/helpers/esm/inheritsLoose\\";
export var Component =
/*#__PURE__*/
function (_Other) {
  _inheritsLoose(Component, _Other);

  function Component() {
    var _this = _Other.call(this) || this;

    _this.flag = true;
    return _this;
  }

  var _proto = Component.prototype;

  _proto.main = function main() {
    alert(this.foo);
  };

  return Component;
}(Other);"
`;

exports[`Transpile: ES5 es2015: for_off 1`] = `
"import \\"core-js/modules/es.symbol\\";
import \\"core-js/modules/es.symbol.description\\";
import \\"core-js/modules/es.symbol.iterator\\";
import \\"core-js/modules/es.array.is-array\\";
import \\"core-js/modules/es.array.iterator\\";
import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.string.iterator\\";
import \\"core-js/modules/web.dom-collections.iterator\\";

function test() {
  for (var _iterator = tasks, _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _iterator[Symbol.iterator]();;) {
    var _ref;

    if (_isArray) {
      if (_i >= _iterator.length) break;
      _ref = _iterator[_i++];
    } else {
      _i = _iterator.next();
      if (_i.done) break;
      _ref = _i.value;
    }

    try {
      executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Transpile: ES5 es2015: generator 1`] = `
"function* idMaker() {
  var index = 0;

  while (true) {
    yield index++;
  }
}"
`;

exports[`Transpile: ES5 es2015: set 1`] = `
"import \\"core-js/modules/es.array.iterator\\";
import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.set\\";
import \\"core-js/modules/es.string.iterator\\";
import \\"core-js/modules/web.dom-collections.iterator\\";
new Set([1, 2, 3]);"
`;

exports[`Transpile: ES5 es2016: exponentiation 1`] = `"var result = Math.pow(2, 3);"`;

exports[`Transpile: ES5 esnext: async 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";

function wait() {
  return new Promise(function ($return, $error) {
    return Promise.resolve(loadData()).then(function () {
      try {
        return $return();
      } catch ($boundEx) {
        return $error($boundEx);
      }
    }, $error);
  });
}"
`;

exports[`Transpile: ES5 esnext: async_loop 1`] = `
"import \\"core-js/modules/es.symbol\\";
import \\"core-js/modules/es.symbol.description\\";
import \\"core-js/modules/es.symbol.iterator\\";
import \\"core-js/modules/es.array.is-array\\";
import \\"core-js/modules/es.array.iterator\\";
import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import \\"core-js/modules/es.string.iterator\\";
import \\"core-js/modules/web.dom-collections.iterator\\";

function executeTasks(tasks) {
  return new Promise(function ($return, $error) {
    var _iterator, _isArray, _i, _ref;

    _iterator = tasks, _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _iterator[Symbol.iterator]();
    var $Loop_2_trampoline;

    function $Loop_2() {
      if (_isArray) {
        if (_i >= _iterator.length) return [1];
        _ref = _iterator[_i++];
      } else {
        _i = _iterator.next();
        if (_i.done) return [1];
        _ref = _i.value;
      }

      var $Try_1_Post = function () {
        try {
          return $Loop_2;
        } catch ($boundEx) {
          return $error($boundEx);
        }
      },
          $Try_1_Catch = function () {
        try {
          console.error(\\"Error\\");
          return $Try_1_Post();
        } catch ($boundEx) {
          return $error($boundEx);
        }
      };

      try {
        return Promise.resolve(executeCommands()).then(function () {
          try {
            return $Try_1_Post();
          } catch ($boundEx) {
            return $Try_1_Catch($boundEx);
          }
        }, $Try_1_Catch);
      } catch (error) {
        $Try_1_Catch(error)
      }
    }

    return ($Loop_2_trampoline = function (q) {
      while (q) {
        if (q.then) return void q.then($Loop_2_trampoline, $error);

        try {
          if (q.pop) {
            if (q.length) return q.pop() ? $Loop_2_exit.call(this) : q;else q = $Loop_2;
          } else q = q.call(this);
        } catch (_exception) {
          return $error(_exception);
        }
      }
    }.bind(this))($Loop_2);

    function $Loop_2_exit() {
      return $return();
    }
  });
}"
`;

exports[`Transpile: ES5 esnext: catch_bind 1`] = `
"try {
  throw 0;
} catch (_unused) {
  doSomethingWhichDoesntCareAboutTheValueThrown();
} finally {
  doSomeCleanup();
}"
`;

exports[`Transpile: ES5 esnext: class properties 1`] = `
"export var Component = function Component() {
  this.onClick = function () {};
};"
`;

exports[`Transpile: ES5 esnext: decorators 1`] = `
"var _class,
    MyClass = annotation(_class = function MyClass() {}) || _class;

function annotation(target) {
  target.annotated = true;
}"
`;

exports[`Transpile: ES5 esnext: dynamic import 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import(
/*webpackChunkName:'lodash-wBYk8'*/
\\"lodash\\").then(function (lodash) {
  console.log(lodash.camelCase(\\"hello world\\"));
});"
`;

exports[`Transpile: ES5 esnext: object spread 1`] = `
"import \\"core-js/modules/es.object.assign\\";
var original = {
  bar: 10
},
    variant = Object.assign({
  foo: 1
}, original);"
`;

exports[`Transpile: ES5 loadable 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import loadable from \\"@loadable/component\\";
export var Home = loadable({
  resolved: {},
  chunkName: function chunkName() {
    return \\"Home-JfPYC\\";
  },
  isReady: function isReady() {
    if (this.resolved[key] === false) {
      return false;
    }

    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[key];
    }

    return false;
  },
  importAsync: function importAsync() {
    return import(
    /* webpackChunkName: \\"Home-JfPYC\\" */
    \\"./views/Home\\");
  },
  requireAsync: function requireAsync() {
    var _this = this,
        key = this.resolve(props);

    this.resolved[key] = false;
    return this.importAsync(props).then(function (resolved) {
      _this.resolved[key] = true;
      return resolved;
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return eval('require.resolve')(\\"./views/Home\\");
  }
});
export var About = loadable({
  resolved: {},
  chunkName: function chunkName() {
    return \\"About-8yDxR\\";
  },
  isReady: function isReady() {
    if (this.resolved[key] === false) {
      return false;
    }

    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[key];
    }

    return false;
  },
  importAsync: function importAsync() {
    return import(
    /* webpackChunkName: \\"About-8yDxR\\" */
    \\"./views/About\\");
  },
  requireAsync: function requireAsync() {
    var _this = this,
        key = this.resolve(props);

    this.resolved[key] = false;
    return this.importAsync(props).then(function (resolved) {
      _this.resolved[key] = true;
      return resolved;
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return eval('require.resolve')(\\"./views/About\\");
  }
});
export var Contact = loadable({
  resolved: {},
  chunkName: function chunkName() {
    return \\"Contact-AIf6m\\";
  },
  isReady: function isReady() {
    if (this.resolved[key] === false) {
      return false;
    }

    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[key];
    }

    return false;
  },
  importAsync: function importAsync() {
    return import(
    /* webpackChunkName: \\"Contact-AIf6m\\" */
    \\"./views/Contact\\");
  },
  requireAsync: function requireAsync(props) {
    var _this = this,
        key = this.resolve(props);

    this.resolved[key] = false;
    return this.importAsync(props).then(function (resolved) {
      _this.resolved[key] = true;
      return resolved;
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return eval('require.resolve')(\\"./views/Contact\\");
  }
});"
`;

exports[`Transpile: ES5 loadable magic-comments 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import loadable from \\"@loadable/component\\";
export var Home = loadable({
  resolved: {},
  chunkName: function chunkName() {
    return \\"Home-JfPYC\\";
  },
  isReady: function isReady() {
    if (this.resolved[key] === false) {
      return false;
    }

    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[key];
    }

    return false;
  },
  importAsync: function importAsync() {
    return import(
    /* webpackChunkName: \\"Home-JfPYC\\" */
    \\"./views/Home\\");
  },
  requireAsync: function requireAsync() {
    var _this = this,
        key = this.resolve(props);

    this.resolved[key] = false;
    return this.importAsync(props).then(function (resolved) {
      _this.resolved[key] = true;
      return resolved;
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return eval('require.resolve')(\\"./views/Home\\");
  }
});
export var About = loadable({
  resolved: {},
  chunkName: function chunkName() {
    return \\"About\\";
  },
  isReady: function isReady() {
    if (this.resolved[key] === false) {
      return false;
    }

    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[key];
    }

    return false;
  },
  importAsync: function importAsync() {
    return import(
    /* webpackChunkName: \\"About\\" */
    \\"./views/About\\");
  },
  requireAsync: function requireAsync() {
    var _this = this,
        key = this.resolve(props);

    this.resolved[key] = false;
    return this.importAsync(props).then(function (resolved) {
      _this.resolved[key] = true;
      return resolved;
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return eval('require.resolve')(\\"./views/About\\");
  }
});
export var Contact = loadable({
  resolved: {},
  chunkName: function chunkName() {
    return \\"Contact-AIf6m\\";
  },
  isReady: function isReady(props) {
    if (this.resolved[key] === false) {
      return false;
    }

    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[key];
    }

    return false;
  },
  importAsync: function importAsync() {
    return import(
    /* webpackMode: \\"lazy\\", webpackFoo: \\"bar\\", webpackChunkName: \\"Contact-AIf6m\\" */
    \\"./views/Contact\\");
  },
  requireAsync: function requireAsync(props) {
    var _this = this,
        key = this.resolve(props);

    this.resolved[key] = false;
    return this.importAsync(props).then(function (resolved) {
      _this.resolved[key] = true;
      return resolved;
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return eval('require.resolve')(\\"./views/Contact\\");
  }
});"
`;

exports[`Transpile: ES5 loadable wildcard 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import \\"core-js/modules/es.regexp.exec\\";
import \\"core-js/modules/es.string.replace\\";
import loadable from \\"@loadable/component\\";
var page = \\"About\\",
    lang = \\"de_DE\\";
export var views = loadable({
  resolved: {},
  chunkName: function chunkName() {
    return (\\"views-\\" + page).replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },
  isReady: function isReady() {
    if (this.resolved[key] === false) {
      return false;
    }

    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[key];
    }

    return false;
  },
  importAsync: function importAsync() {
    return import(
    /* webpackChunkName: \\"views-[request]\\" */
    \\"./views/\\" + page);
  },
  requireAsync: function requireAsync() {
    var _this = this,
        key = this.resolve(props);

    this.resolved[key] = false;
    return this.importAsync(props).then(function (resolved) {
      _this.resolved[key] = true;
      return resolved;
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/\\" + page);
    }

    return eval('require.resolve')(\\"./views/\\" + page);
  }
});
export var leanIntlData = loadable({
  resolved: {},
  chunkName: function chunkName() {
    return (\\"lean-intl-locale-data-\\" + lang).replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },
  isReady: function isReady() {
    if (this.resolved[key] === false) {
      return false;
    }

    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[key];
    }

    return false;
  },
  importAsync: function importAsync() {
    return import(
    /* webpackChunkName: \\"lean-intl-locale-data-[request]\\" */
    \\"./lean-intl/locale-data/\\" + lang);
  },
  requireAsync: function requireAsync() {
    var _this = this,
        key = this.resolve(props);

    this.resolved[key] = false;
    return this.importAsync(props).then(function (resolved) {
      _this.resolved[key] = true;
      return resolved;
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./lean-intl/locale-data/\\" + lang);
    }

    return eval('require.resolve')(\\"./lean-intl/locale-data/\\" + lang);
  }
});
export var reactIntlData = loadable({
  resolved: {},
  chunkName: function chunkName() {
    return (\\"react-intl-locale-data-\\" + lang).replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },
  isReady: function isReady() {
    if (this.resolved[key] === false) {
      return false;
    }

    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[key];
    }

    return false;
  },
  importAsync: function importAsync() {
    return import(
    /* webpackChunkName: \\"react-intl-locale-data-[request]\\" */
    \\"./react-intl/locale-data/\\" + lang);
  },
  requireAsync: function requireAsync(props) {
    var _this = this,
        key = this.resolve(props);

    this.resolved[key] = false;
    return this.importAsync(props).then(function (resolved) {
      _this.resolved[key] = true;
      return resolved;
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./react-intl/locale-data/\\" + lang);
    }

    return eval('require.resolve')(\\"./react-intl/locale-data/\\" + lang);
  }
});"
`;

exports[`Transpile: ES5 lodash: async 1`] = `
"import _parallel from \\"async/parallel\\";

_parallel([]);"
`;

exports[`Transpile: ES5 lodash: lodash 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

_camelCase(\\"hello world\\");"
`;

exports[`Transpile: ES5 lodash: recompose 1`] = `
"import _compose from \\"recompose/compose\\";

_compose([]);"
`;

exports[`Transpile: ES5 macro: gql 1`] = `
"var query = {
  kind: \\"Document\\",
  definitions: [{
    kind: \\"OperationDefinition\\",
    operation: \\"query\\",
    name: {
      kind: \\"Name\\",
      value: \\"User\\"
    },
    variableDefinitions: [],
    directives: [],
    selectionSet: {
      kind: \\"SelectionSet\\",
      selections: [{
        kind: \\"Field\\",
        name: {
          kind: \\"Name\\",
          value: \\"user\\"
        },
        arguments: [{
          kind: \\"Argument\\",
          name: {
            kind: \\"Name\\",
            value: \\"id\\"
          },
          value: {
            kind: \\"IntValue\\",
            value: \\"5\\"
          }
        }],
        directives: [],
        selectionSet: {
          kind: \\"SelectionSet\\",
          selections: [{
            kind: \\"Field\\",
            name: {
              kind: \\"Name\\",
              value: \\"lastName\\"
            },
            arguments: [],
            directives: []
          }, {
            kind: \\"FragmentSpread\\",
            name: {
              kind: \\"Name\\",
              value: \\"UserEntry1\\"
            },
            directives: []
          }]
        }
      }]
    }
  }],
  loc: {
    start: 0,
    end: 79,
    source: {
      body: \\"\\\\n  query User {\\\\n    user(id: 5) {\\\\n      lastName\\\\n      ...UserEntry1\\\\n    }\\\\n  }\\\\n\\",
      name: \\"GraphQL request\\",
      locationOffset: {
        line: 1,
        column: 1
      }
    }
  }
};"
`;

exports[`Transpile: ES5 macro: preval 1`] = `"var one = 1;"`;

exports[`Transpile: ES5 promise 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
export default function () {
  return new Promise(function (resolve) {
    return resolve(42);
  });
}"
`;

exports[`Transpile: ES5 react: jsx 1`] = `
"function render() {
  return React.createElement(\\"h1\\", null, \\"Hello\\");
}"
`;

exports[`Transpile: ES5 react: jsx spread 1`] = `
"import \\"core-js/modules/es.object.assign\\";

function render(props) {
  return React.createElement(\\"h1\\", Object.assign({
    className: \\"large\\"
  }, props), \\"Hello\\");
}"
`;

exports[`Transpile: ES5 require context 1`] = `
"if (process.env.NODE_ENV === \\"test\\") {
  require(\\"babel-plugin-require-context-hook/register\\")();
}

var loader = require.context(__dirname, false, /\\\\.js/);"
`;

exports[`Transpile: ES5 typescript.tsx 1`] = `
"export var hasPackageConfig = function hasPackageConfig(name) {
  try {
    return !!getPackageConfig()[name];
  } catch (_) {
    return false;
  }
};"
`;
